docs/registering-a-control-handler-function.md

```diff
@@ -90,3 +93,135 @@ int main(void)
     return 0;
 }
 ```
+
+## Listen with Hidden Window Example
+
+Per the remarks, if the gdi32.dll or user32.dll library are loaded,  **SetConsoleCtrlHandler** does not get called for the **CTRL\_LOGOFF\_EVENT** and **CTRL\_SHUTDOWN\_EVENT** events. The workaround specified is to create a hidden window by calling the [**CreateWindowEx**](/windows/win32/api/winuser/nf-winuser-createwindowexa) method with the *dwExStyle* parameter set to 0 and listen for the  [**WM\_QUERYENDSESSION**](/windows/win32/shutdown/wm-queryendsession) and [**WM\_ENDSESSION**](/windows/win32/shutdown/wm-endsession) window messages.
+
+It is also valid to have a visible window and listen for these messages. It is likely that in using gdi32.dll or user32.dll that the application would have a window. In that case, add the two messages to the existing [Window Procedure](/windows/win32/winmsg/using-window-procedures).
```

Comment:
You can remove this sentence if you lead this section with "create a hidden window, if no window already exists" or similar.

Reply:
Condensed.